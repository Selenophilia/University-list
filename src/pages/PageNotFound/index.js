import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import notExist from "../../assets/images/404.png";
import { useHistory } from "react-router-dom";
import { Button } from "@material-ui/core";

const useStyles = makeStyles({
  container: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
    height: "70vh"
  },
  home: {
    backgroundColor: "#3CB371",
    color: "#FFFFFF",
    marginTop: 50,
    width: 260,
    "&:hover": {
      backgroundColor: "#38a167"
    }
  }
});

const PageNotFound = () => {
  const history = useHistory();
  const classes = useStyles();
  return (
    <div className={classes.container}>
      <img src={notExist} />
      <Button
        variant="contained"
        className={classes.home}
        onClick={() => history.push("/")}
      >
        {"Go back to home page"}
      </Button>
    </div>
  );
};

export default PageNotFound;
