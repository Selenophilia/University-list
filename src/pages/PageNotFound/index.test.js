import React from "react";
import { render, screen } from "@testing-library/react";
import PageNotFound from ".";

const renderRoot = () => {
  return render(<PageNotFound />);
};

it("renders PageNotFound component correctly ", () => {
  const { asFragment } = renderRoot();
  expect(asFragment()).toMatchSnapshot();
});

it("calls home page button in page", () => {
  renderRoot();
  expect(screen.getByTestId("home")).toBeInTheDocument();
});
