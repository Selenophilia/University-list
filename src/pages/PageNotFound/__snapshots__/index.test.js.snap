// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PageNotFound component correctly  1`] = `
<DocumentFragment>
  <div
    class="makeStyles-container-1"
  >
    <img
      src="404.png"
    />
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-home-2"
      data-testid="home"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Go back to home page
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
